drivers/sensor/lsm6dsr/lsm6dsr.h

```diff
@@ -0,0 +1,98 @@
+/* ST Microelectronics LSM6DSO 6-axis IMU sensor driver
+ *
+ * Copyright (c) 2019 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Datasheet:
+ * https://www.st.com/resource/en/datasheet/lsm6dsr.pdf
+ */
+
+#ifndef ZEPHYR_DRIVERS_SENSOR_LSM6DSR_LSM6DSR_H_
+#define ZEPHYR_DRIVERS_SENSOR_LSM6DSR_LSM6DSR_H_
+
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/types.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/spi.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/util.h>
+#include <stmemsc.h>
+#include "lsm6dsr_reg.h"
+#include <zephyr/drivers/spi.h>
+#include <zephyr/drivers/i2c.h>
+
+#define LSM6DSR_EN_BIT					0x01
+#define LSM6DSR_DIS_BIT					0x00
+
+/* Accel sensor sensitivity grain is 61 ug/LSB */
+#define SENSI_GRAIN_XL				(61LL)
+
+/* Gyro sensor sensitivity grain is 4375 udps/LSB */
+#define SENSI_GRAIN_G				(4375LL)
+#define SENSOR_PI_DOUBLE			(SENSOR_PI / 1000000.0)
+#define SENSOR_DEG2RAD_DOUBLE		(SENSOR_PI_DOUBLE / 180)
+#define SENSOR_G_DOUBLE				(SENSOR_G / 1000000.0)
+
+struct lsm6dsr_config {
+	stmdev_ctx_t ctx;
+	union {
+		const struct i2c_dt_spec i2c;
+		const struct spi_dt_spec spi;
+	} stmemsc_cfg;
+	uint8_t accel_pm;
+	uint8_t accel_odr;
+	uint8_t accel_range;
+	uint8_t gyro_pm;
+	uint8_t gyro_odr;
+	uint8_t gyro_range;
+	bool drdy_pulsed;
+#ifdef CONFIG_LSM6DSR_TRIGGER
+	const struct gpio_dt_spec gpio_drdy;
+	uint8_t int_pin;
+	bool trig_enabled;
+#endif
+};
+
+struct lsm6dsr_data {
+	const struct device *dev;
+	int16_t acc[3];
+	uint32_t acc_gain;
+	uint8_t accel_pm;
+	int16_t gyro[3];
+	uint32_t gyro_gain;
+	int16_t temp_sample;
+
+	float accel_freq;
+	uint8_t accel_fs;
+	float gyro_freq;
+	uint8_t gyro_fs;
```

Comment:
looks like there's room here for mem alignment optimization to minimize holes

Reply:
I think is fixed now.